woocommerce/woocommerce-android
# Added a basic screenshot test, lane and some new mocks. PR 1 of 2.

### Description
Out of 16 files changed, 9 are mocks.

This PR combines some parts of existing work on screenshots from [current repo, but old branch](https://github.com/woocommerce/woocommerce-android/compare/add/screenshots-after-dark-mode) (h/t Gio) and screenshots-related code from [WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/tree/trunk/fastlane).

1. `ScreenshotTest.kt` was extended with actual steps that will go through app areas and capture screenshots, and save them to Emulator storage.
2. Dark/Light theme toggling is done by a test, inside the application, because [toggling it via `abd`](https://android.stackexchange.com/questions/232271/setting-android-10-dark-theme-on-and-off-with-adb-shell-how-to-do-it-without-r) still proves to be not working.
3. A new `take_screenshots` lane is added to `Fastfile`. It's very basic. No support for promo screenshots or Emulators images handling so far.
4. Since locale switching does not work ([fastlane bug](https://github.com/fastlane/fastlane/issues/19521)), the lane will currently use only one hardcoded locale (`en-EN`). In fact, the value of hardcoded locale does not matter, because switch will not take place anyway.
5. Some mocks were added and updated, to make the screenshots content more pleasant to the eye.

This is the first PR of expected two. For the next PR, it's planned to add the promo screenshots generation logic: embedding the pictures into a device frame and adding promo strings next to the pictures, as can be seen on [Play Store](https://play.google.com/store/apps/details?id=com.woocommerce.android), and hopefully some Emulators handling logic.

### Testing instructions
1. Make sure the Emulator is running. Emulator startup/creation is not currently handled by the lane. It's preferable to use `Pixel 2 XL API 28`.
2. Make sure that `adb` is added to the PATH. This can for example be done by adding `export PATH="/Users/<user>/Library/Android/sdk/platform-tools/":$PATH` as a first line of `~/.bash_profile` and then running `source ~/.bash_profile`.
3. Run `bundle install`
4. Run `bundle exec fastlane take_screenshots`

Fastlane should assemble the Vanilla variants of the app, this takes some time for the first execution, and then should run the screenshot test twice: for dark and for light themes. At the end, a new `screenshots` folder should be created in repo `fastane` folder, with a total of 16 pictures (8 for each theme).

5. You can manually set the emulator language to some other, to make sure it will still work. I tried only Italian so far, so I'm not sure if it will work for _all_ languages from MAG16.
6. I've tried to manupulate mocks data a little, to have the screenshots close to pictures from [WC Play Store page](https://play.google.com/store/apps/details?id=com.woocommerce.android). Any comments about that are welcomed too.
